pkg/util/version/stream.go

```diff
@@ -13,17 +13,19 @@ type Stream struct {
 
 // GetStream return matching stream, used to upgrade cluster.
 func GetStream(v *Version) (*Stream, error) {
+	// Try to match X.Y
 	for _, s := range Streams {
 		if s.Version.V[0] == v.V[0] &&
 			s.Version.V[1] == v.V[1] {
```

Comment:
I am finding it really difficult to mentally parse the logic of this code and evaluate the change. I've spent about 30m trying to type up what this currently does and what it should do with little success. Hence, I think we need some refactoring to improve the legibility of this code.

First, I propose we do a quick refactor of the `Version` struct to use "major", "minor", and "patch" version fields rather than trying to interpret "0", "1", and "2". That would have made reading this code a bit easier.

Second, I propose we improve the naming and comments. The purpose of this code is to determine if a valid upgrade stream `s` is available for a cluster of a given version `v`, but that's not immediately clear. For the function signature, I have included a review comment above with a suggestion for improving this. Within the scope of the function, `v` and `s` don't convey the purpose of each version. Let's consider using e.g. `targetVersion` and `upgradeVersionCandidate`?

And last, we need to consider our use case and what we are trying to accomplish. The RP supports initiating upgrades via the CVO for at most two streams, but in this PR we're removing the second one, so we're only really considering one possible target upgrade stream. How often are we even going to care about supporting 2 upgrade streams within the RP? If we jump back and forth, are we going to need a code update every time? The logic as written, given a single upgrade stream, is sort of redundant and vestigial. I'd rather change the code back to only accepting a single stream, or write something that gives me more confidence for the two stream case, than write something like this where we're anticipating >2 streams but not actually testing a greater number of stream combinations because we don't really care about that use case.

Reply:
Feel free to open follow-up issue for these reactors. I don't see this as a priority at this point. 
And this will be way bigger refactor at this point which I want to avoid.